src/screens/Feed.js

```diff
@@ -1,124 +1,122 @@
 import ProfileAvatar from './app/Components/ProfileAvatar/ProfileAvatar';
 import Post from './app/Components/Post/Post';
+import Swiper from 'react-native-swiper';
+import {Divider} from 'react-native-paper';
+import firestore from '@react-native-firebase/firestore';
+import * as ScreenshotDetector from 'react-native-screenshot-detect';
+
 export default class FeedScreen extends React.Component {
   constructor(props) {
     super(props);
-    this.state = {};
+    this.state = {
+      swiperIndex: 0,
+      posts: [],
+      loading: false,
+    };
+  }
+
+  componentDidMount() {
+    this.fetchPosts();
+
+    this.eventEmitter = ScreenshotDetector.subscribe(() => {
+      Alert.alert(
+        'The post with caption',
+        posts[this.state.swiperIndex].caption,
+        'was screenshotted',
+      );
+    });
+  }
+
+  componentWillUnmount() {
+    ScreenshotDetector.unsubscribe(this.eventEmitter);
   }
 
+  fetchPosts = () => {
+    const subscriber = firestore()
+      .collection('posts')
+      .onSnapshot(querySnapshot => {
+        const posts = [];
+
+        querySnapshot.forEach(documentSnapshot => {
+          posts.push({
+            ...documentSnapshot.data(),
+            key: documentSnapshot.id,
+          });
+        });
+
+        this.setState({
+          posts,
+          loading: false,
+        });
+      });
+  };
+
   // renders each post function. Each post is a card with a cover(the image) and content. Content for now has a caption with the username and 2 buttons
   renderPosts = element => {
     const {
-      img,
+      allowScreenshot,
       caption,
+      dateCreated,
+      hearts,
+      image,
+      key,
+      uid,
       username,
-      comments,
-      likes,
-      likers,
-      bookmarkers,
-    } = element.item;
-    const uid = 'dennyliang'; // this is the acc im logged on
-    const isLiked = likers[uid] === true ? true : false; // if i liked it
-    const isBookmarked = bookmarkers[uid] === true ? true : false;
+    } = element;
+    // const isLiked = likers[uid] ? true : false; // if i liked it
+    // const isBookmarked = bookmarkers[uid] ? true : false;
+
     return (
-      <View>
-        <View style={styles.topBarViewStyles}>
-          <ProfileAvatar username={username} img={img} />
-          <SettingsButton icon="horizontal" />
+      <TouchableWithoutFeedback
+        key={key}
+        onPress={() => this.props.navigation.navigate('UserProfile')}>
+        <View>
+          <View style={styles.topBarViewStyles}>
+            <ProfileAvatar username={username} img={image} />
+            <SettingsButton icon="horizontal" />
+          </View>
+          <Post
+            img={image}
+            isLiked={true}
+            likes={hearts}
+            isBookmarked={true}
+            caption={caption}
+            username={username}
+            commentsLength={0}
+          />
+          <Divider />
```

Comment:
Need to go back to fetching likes and bookmarks from database, was this ever completed? Please keep isLiked and isBookarmed as dynamic variables so we can edit in the future